tensorly/contrib/decomposition/tt_TTOI.py

```diff
@@ -0,0 +1,255 @@
+import numpy as np
+from math import prod
+import tensorly as tl
+from tensorly.tt_tensor import tt_to_tensor, validate_tt_rank
+import warnings
+from tensorly.decomposition._base_decomposition import DecompositionMixin
+
+
+def sequential_prod(tensor_prod, multiplier_list, left_to_right=True):
+    """Perform sequential multiplication and reshaping
+
+    Parameters
+    ----------
+    tensor_prod: ndrray
+        the tensor to be multiplied by a list of tensors
+        tensor_prod is of dimension r_0*p_1*p_2*...*p_d*r_{d-1}
+    multiplier_list: list 
+        a list of tensors to multiply the tensor
+        len(multiplier_list) <= d-1
+        If direction == "left", multiplier_list[i] is of shape (r_i, p_{i+1}, r_{i+1})
+        If direction == "right", multiplier_list[i] is of shape (r_{d-i}, p_{d-i}, r_{d-i-1})
+    left_to_right : bool, optional, default is True
+        direction of the multiplication
+    Returns
+    -------
+    tensor_prod : ndarray
+        product tensor
+        If direction == "left" and `multiplier_list` contains N tensors, then `tensor_prod` is of dimension r_N*p_{N+1}*...*p_d*r_{d-1}
+        If direction == "right" and `multiplier_list` contains d-N tensors, then `tensor_prod` is of dimension r_0*p_1*...*p_N*r_N
+    """
+
+    if left_to_right == True:
+        for i in range(len(multiplier_list)):
+            tensor_prod = tl.tensordot(
```

Comment:
Just wondering if instead of tensordot, it would be possible to use the n_mode_product which, in my experience of Tensorly, is often faster. Tensordot calls the backend's tensordot directly with n_mode_product might benefit from any improved implementaton of tensor contractions we might add to Tensorly in the future.

Reply:
It seems that the n_mode_product function is for tensor-matrix product or tensor-vector product; is that correct? I am using tensordot here for sequential tensor product, and hence would prefer not to change this function if possible....

Reply:
You are right, ignore my comment!